state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java

```diff
@@ -28,25 +29,34 @@
 @RequiredArgsConstructor
 public class DefaultApolloClientFactory implements ApolloClientFactory {
   @NonNull private final String streamRegistryUrl;
-  @NonNull private final Consumer<ApolloClient.Builder> configurer;
+  private Credentials credentials;
 
-  public DefaultApolloClientFactory(String streamRegistryUrl) {
-    this(streamRegistryUrl, builder -> {});
+  public DefaultApolloClientFactory(String streamRegistryUrl, Credentials credentials) {
+    this(streamRegistryUrl);
+    this.credentials = credentials;
   }
 
   @Override
   public ApolloClient create() {
-    var builder = builder()
-        .okHttpClient(new OkHttpClient.Builder().build());
-    configurer.accept(builder);
-    return builder
+    var okHttpClientBuilder = new OkHttpClient.Builder();
+    if (credentials != null) {
+      okHttpClientBuilder.addInterceptor(chain -> chain.proceed(chain.request().newBuilder()
+          .header("Authorization", "Basic " + Base64.getEncoder().encodeToString((credentials.getUserName() + ":" + credentials.getPassword()).getBytes(UTF_8)))
```

Comment:
I'm not seeing any evidence that this is actually accepted by the server.
Perhaps that isn't needed, is the server handling this yet?

Approved with this reservation.